source-pendo/source_pendo/api.py

```diff
@@ -0,0 +1,123 @@
+import json
+from logging import Logger
+from typing import AsyncGenerator, Iterable
+
+from estuary_cdk.capture.common import LogCursor
+from estuary_cdk.http import HTTPSession
+
+from .models import Event, EventResponse, Resource, Metadata
+
+API = "https://app.pendo.io/api/v1"
+
+
+async def fetch_resources(
+        http: HTTPSession,
+        entity: str,
+        log: Logger,
+) -> Iterable[Resource]:
+    url = f"{API}/{entity}"
+
+    response = json.loads(await http.request(log, url))
+
+    resources: list[Resource] = []
+    for id, resource in enumerate(response):
+        r = Resource.model_validate_json(json.dumps(resource))
+        r.meta_ = Resource.Meta(op="c", row_id=id)
```

Comment:
For snapshot bindings, I don't think we should need to or want to set the metadata manually like this, since it should handled here: https://github.com/estuary/connectors/blob/602f4bb3a139f944dfad68cff890bf98b8771759/estuary-cdk/estuary_cdk/capture/common.py#L523-L525